apps/condo/domains/resident/tasks/helpers/sendBillingReceiptsAddedNotifications.js

```diff
@@ -95,27 +99,25 @@ const prepareAndSendNotification = async (context, receipt, resident) => {
  */
 const sendBillingReceiptsAddedNotificationsForPeriod = async (receiptsWhere, onLastDtChange) => {
     const { keystone: context } = await getSchemaCtx('Resident')
-    const receiptsCount = await BillingReceipt.count(context, receiptsWhere)
     let skip = 0, successCount = 0
     const notifiedUsers = new Set()
 
-    logger.info({ msg: 'sending billing receipts', receiptsCount, data: receiptsWhere })
-
-    // Exit if no receipts found to proceed
-    if (!receiptsCount) return
-
-    let lastReceipt
-
-    while (skip < receiptsCount) {
-        const receipts = await BillingReceipt.getAll(context, receiptsWhere, { sortBy: ['createdAt_ASC'], first: CHUNK_SIZE, skip })
+    let lastReceipt, currBatchLength = CHUNK_SIZE
 
+    while (currBatchLength === CHUNK_SIZE) {
+        const receipts = await BillingReceipt.getAll(
```

Comment:
maybe it is best to 

Get new batch IDs using getAll to calculate access / use GqlLoadKnexList or smth like this